core/libp2p/peer/peer_id.cpp

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/peer/peer_id.hpp"
+
+#include <boost/format.hpp>
+#include "crypto/sha/sha256.hpp"
+#include "libp2p/multi/multihash.hpp"
+
+namespace libp2p::peer {
+  using Encodings = multi::MultibaseCodec::Encoding;
+
+  PeerId::PeerId(kagome::common::Buffer id,
+                 const multi::MultibaseCodec &multibase_codec,
+                 const crypto::CryptoProvider &crypto_provider)
```

Comment:
PeerId is an Entity. You should read this https://github.com/soramitsu/kagome/blob/master/docs/dev-guide.md

Entity MUST NOT depend on the outer layer (interface). Instead, you must create new interface + impl - PeerIdFactory which will instantiate PeerId

Reply:
Decided to rename existing Factory to Manager and delegate some of the methods of PeerId to the Manager, so that no dependencies will exist